homework/solutions/nelson/session1-spider1.py

```diff
@@ -0,0 +1,102 @@
+from asyncio.futures import Future
+from asyncio.tasks import sleep
+import logging
+from typing import List
+import asyncio
+import sys
+
+import aiohttp
+from typing_extensions import IntVar
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+async def get(url: str = None, counter: int = 0, callback = None, session: aiohttp.ClientSession = None) -> str:
+    """
+    Get a webpage's content, given it's URL. Uses only a simple GET HTTP Request
+
+    :param str url: A fully qualified URL to fetch.
+    :param int counter: Which call is this? Order matters!
+    :param aiohttp.ClientSession session: A re-usable session for making calls.
+
+    :return: str The result.
+    """
+    if session is None:
+        raise ValueError('`session` is a required argument. You must pass an HTTP Session')
+    logger.info(f'Awaiting {url} in call {counter}.')
+    result = await session.get(url)
+    if callback:
+        callback(url, counter, await result.text())
+    return await result.text()
+
+def get_using_task(url: str, counter: int, callback = None, session : aiohttp.ClientSession = None) -> asyncio.Task:
+    if session is None:
+        raise ValueError('`session` is a required argument. You must pass an HTTP Session')
+
+    mycoroutine = get(url, counter, callback, session)
+    return asyncio.create_task(mycoroutine)
+
+async def spider(urls: List[str], concurrency: int = 1, callback = None, session : aiohttp.ClientSession = None):
+    if session is None:
+        raise ValueError('`session` is a required argument. You must pass an HTTP Session')
+
+    complete_tasks = set()
```

Comment:
How come you're creating `complete_tasks` and `pending_tasks` when they're already provided by `asyncio.wait`?